frame/democracy/src/lib.rs

```diff
@@ -645,10 +569,14 @@ pub mod pallet {
 			}
 
 			T::Currency::reserve(&who, value)?;
+
+			let depositors = BoundedVec::<_, T::MaxDeposits>::truncate_from(vec![who.clone()]);
+			DepositOf::<T>::insert(index, (depositors, value));
+
 			PublicPropCount::<T>::put(index + 1);
-			<DepositOf<T>>::insert(index, (&[&who][..], value));
 
-			<PublicProps<T>>::append((index, proposal_hash, who));
+			let ok = <PublicProps<T>>::try_append((index, proposal, who)).is_ok();
+			debug_assert!(ok, "have checked real_prop_count < max_proposals; qed");
```

Comment:
we can just return an error here now a days

Reply:
Meh - the logic works as it is and doing this will complicate test code as the function will need to be called explicitly through a transactional wrapper.

Reply:
Tests are now also using transactional.